src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,549 @@
+                    dns.write(dns_check + "\nWe found an issue with the DNS resolution on your cluster.")
+            return False, storage_space_available
+
+        else:
+            if storage_space_available :
+                dns_check_path = os.path.join(filepath_with_timestamp, "DNS_Check.txt")
+                with open(dns_check_path, 'w+') as dns:
+                    dns.write(dns_check + "\nCluster DNS check passed successfully.")
+            return True, storage_space_available
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+
+    except Exception as e:
+        print(e)
+    
+    return True, storage_space_available
+
+
+def check_cluster_outbound_connectivity(diagnoser_container_log, filepath_with_timestamp, storage_space_available):
+
+    try:
+
+        # To retreive only the outbound connectivity result from the diagnoser container logs
+        outbound_check = diagnoser_container_log[-4:-1:]
+
+        # Validating if outbound connectiivty is working or not and displaying proper result
+        if(outbound_check != "000"):
+            if storage_space_available :
+                outbound_connectivity_check_path = os.path.join(filepath_with_timestamp, "Outbound_Network_Connectivity_Check.txt")
+                with open(outbound_connectivity_check_path, 'w+') as outbound:
+                    outbound.write("Response code " + outbound_check + "\nOutbound network connectivity check passed successfully.")
+            return True, storage_space_available
+        else:
+            print("Error: We found an issue with outbound network connectivity from the cluster.\nIf your cluster is behind an outbound proxy server, please ensure that you have passed proxy paramaters during the onboarding of your cluster.\nFor more details visit 'https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli#connect-using-an-outbound-proxy-server'.\nPlease ensure to meet the following network requirements 'https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/quickstart-connect-cluster?tabs=azure-cli#meet-network-requirements' \n")
+            if storage_space_available :
+                outbound_connectivity_check_path = os.path.join(filepath_with_timestamp, "Outbound_Network_Connectivity_Check.txt")
+                with open(outbound_connectivity_check_path, 'w+') as outbound:
+                    outbound.write("Response code " + outbound_check + "\nWe found an issue with Outbound network connectivity from the cluster.")
+            return False, storage_space_available
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+
+    except Exception as e:
+        print(e)
+    
+    return True, storage_space_available
+
+def check_msi_certificate(corev1_api_instance):
```

Comment:
is it needed to pass corev1_Api_instance as parameter?

Reply:
Yes its required to fetch the secrets which is used to check for msi certificate present or not.